driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -945,26 +945,13 @@ void tryHandOverOrRelease(final UsageTrackingInternalConnection openConnection)
             }
         }
 
-        /**
-         * @param timeout If {@linkplain Timeout#isInfinite() infinite},
-         *                then the lock is {@linkplain Lock#lockInterruptibly() acquired interruptibly}.
-         */
-        private void tryLock(final Timeout timeout) throws MongoTimeoutException {
-            boolean success;
+        private void lockInterruptibly(final Lock lock) throws MongoTimeoutException {
```

Comment:
You can remove the throws clause now.

Reply:
Thanks, it should have been `MongointerruptedException`. Done.